test_pull_requests

```diff
@@ -1235,31 +1441,30 @@ popd
               create_or_update_comment(id, base_repo, ACTION_PREFIX, ACTION_NOT_TEAM, comments)
             end
           else
-            updated_comment = "#{settings['test_prefix']} Waiting for stable build of '#{branch_settings(branch, settings)['downstream_job_name']}'"
+            updated_comment = compose_bot_comment(settings['test_prefix'], :content => waiting_for_stable_build(branch, settings), :state => :wait_stable_build)
           end
 
-        when /^#{settings['test_prefix']} Waiting/
-          # In this case, we are in one of two states:
+        when :wait_stable_build, :wait_queue_pos
+          # In this case, we are in one of three states:
           #  1) waiting for a downstream project in Jenkins to be stable, so we can begin tests
-          #  2) waiting in a queue to begin tests
+          #  2) determining build queue pos
           #
-          # Two state transitions are possible from state 1):
-          #  - loop back into state 1) if the Jenkins project is unstable
+          # Two state transitions are possible from state 1 and 2):
+          #  - loop back into state 1 (or 2) if the Jenkins project is unstable
           #  - into the 'running tests' state if the Jenkins project is stable
           # If the main pull request or any of the co-requisites have been
           # updated since the last time we evaluated the main pull request, we need to re-queue the
           # build and test.
           #
-          # No state transitions occur from state 2)
-          # TODO: currently, we have two invalid states that will make it through this logic: the
-          # cases where we are not in the build queue but have submitted tests, and the cases where
-          # we are in the build queue but have not submitted tests. Neither of these cases are valid
-          # and we need to reconsider why they're tolerated
+          # TODO: currently, we have one invalid state that will make
```

Comment:
I am not sure it was ever a case we could hit (if I am reading it correctly).  It was an argument of a theoretical case on the matrix of possible states that wouldn't be handled.